### PR TITLE
connect - implement device translations

### PR DESCRIPTION
**Pull request by mroz22**
## Add changeLanguage to @trezor/connect 

- updated submodules
- updated protobuf
- type improvements in fetch assets util
- added TrezorConnect.changeLanguage method
- connect-explorer updated
- `device.availableTranslations` field added

notes for implementation in suite (@komret):
- I don't know what UX in suite should be like. Is it going to be somehow interconnected with firmware update? cc @Hermez-cz 
- there is `device.availableTranslations` field which tells you what translations are available
- this branch will take a while to finish so please build on top of this.
- so far I am not considering doing this automatically as part of firmware update in connect, but I am willing to negotiate
- you can play around with it in explorer https://suite.corp.sldev.cz/connect/connect/translations/#/method/changeLanguage

image

image

error when wrong baseUrl  
image

part of #10099, #11234

### CODE REVIEW

File: packages/protobuf/messages.json [hunk 1]
```diff
@@ -4638,6 +4641,42 @@
                 }
             }
         },
+        "ChangeLanguage": {
```
Comment (mroz22) [inline packages/protobuf/messages.json]: silent update options:
- outadated language blob
- wiped device
- fw upgrade flow

File: packages/connect/src/api/changeLanguage.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        if (binary) {
+            return this.uploadTranslationData(binary);
+        }
+
+        const version = this.device.getVersion().join('.');
+        const model = this.device.features.internal_model;
+
+        // todo: signed?
+        const url = `${baseUrl}/data/translations/translation-${model}-${language}-${version}-unsigned.bin`;
```
Comment (mroz22) [inline packages/connect/src/api/changeLanguage.ts:94]: I am not sure yet if we want to sometimes work with signed and sometimes with unsigned. maybe for development purposes we are going to need unsigned? so maybe `signed` vs `unsigned` should become param of this connect method? 
cc @Hannsek

**Comment by mroz22** [comment]
more or less ready to be looked on. I'd like to appoint @martykan as a reviewer please 🙏

File: packages/connect/src/api/changeLanguage.ts (hunk 2 above)
Comment (Hannsek) [reply packages/connect/src/api/changeLanguage.ts:94]: Maybe? But maybe it is not needed? What do you think @matejcik @grdddj ?

File: packages/connect/src/types/api/changeLanguage.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Type, Static } from '@trezor/schema-utils';
+import type { Params, Response } from '../params';
+import { PROTO } from '../../constants';
+
+export type ChangeLanguage = Static<typeof ChangeLanguage>;
+
+// todo: is this the right way to define discriminated unions?
```
Comment (martykan) [inline packages/connect/src/types/api/changeLanguage.ts]: Yes

File: packages/connect-explorer/src/data/methods/management/firmwareUpdate.ts [hunk 4]
```diff
@@ -17,30 +12,6 @@ export default [
                 optional: false,
                 type: 'file',
             },
-            {
```
Comment (martykan) [inline packages/connect-explorer/src/data/methods/management/firmwareUpdate.ts:20]: Just curious why was this removed?

File: packages/connect/src/device/Device.ts [hunk 5]
```diff
@@ -579,6 +583,7 @@ export class Device extends TypedEmitter<DeviceEvents> {
 
         this.name = deviceInfo.name;
 
+        // todo: move to 553
```
Comment (martykan) [inline packages/connect/src/device/Device.ts:586]: What for?

File: packages/connect-explorer/src/data/methods/management/firmwareUpdate.ts (hunk 4 above)
Comment (mroz22) [reply packages/connect-explorer/src/data/methods/management/firmwareUpdate.ts:20]: unerelated commit. noticed this unused code so I removed it

File: packages/connect/src/device/Device.ts (hunk 5 above)
Comment (mroz22) [reply packages/connect/src/device/Device.ts:586]: there is a block that executes only if new version differs from previous version. imho this feature falls into the category of features which may not change unless firmware version changes too 
image

but I didn't want to make such unerelated runtime change here, so I put a comment.

Comment (martykan) [reply packages/connect/src/device/Device.ts:586]: Ok

**Review by martykan — APPROVED** [review]
